lua/oil/init.lua

```diff
@@ -908,16 +909,21 @@ end
 
 local function close_preview_window_if_not_in_oil()
   local util = require("oil.util")
+  local preview_win_id = util.get_preview_win()
+  if not preview_win_id or not vim.w[preview_win_id].oil_entry_id then
+    return
+  end
 
-  if not util.is_oil_bufnr(0) then
-    local preview_win_id = util.get_preview_win()
-    if preview_win_id then
-      local current_win_id = vim.api.nvim_get_current_win()
-      if current_win_id and current_win_id ~= preview_win_id then
-        vim.api.nvim_win_close(preview_win_id, true)
-      end
+  local oil_source_win = vim.w[preview_win_id].oil_source_win
+  if oil_source_win and vim.api.nvim_win_is_valid(oil_source_win) then
+    local src_buf = vim.api.nvim_win_get_buf(oil_source_win)
+    if vim.bo[src_buf].filetype == "oil" then
```

Comment:
could we use `util.is_oil_bufnr` instead of checking `filetype`?

Reply:
sure yeah, I forgot that utility existed